frontend/src/app/shared/components/storages/file-picker-base-modal.component.ts/file-picker-base-modal.component.ts

```diff
@@ -0,0 +1,153 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See COPYRIGHT and LICENSE files for more details.
+//++
+
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  Inject,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+
+import { IHalResourceLink } from 'core-app/core/state/hal-resource';
+import { IStorageFile } from 'core-app/core/state/storage-files/storage-file.model';
+import { OpModalLocalsMap } from 'core-app/shared/components/modal/modal.types';
+import { OpModalComponent } from 'core-app/shared/components/modal/modal.component';
+import { OpModalLocalsToken } from 'core-app/shared/components/modal/modal.service';
+import { SortFilesPipe } from 'core-app/shared/components/storages/pipes/sort-files.pipe';
+import { StorageFilesResourceService } from 'core-app/core/state/storage-files/storage-files.service';
+import { Breadcrumb, BreadcrumbsContent } from 'core-app/spot/components/breadcrumbs/breadcrumbs-content';
+import {
+  StorageFileListItem,
+} from 'core-app/shared/components/storages/storage-file-list-item/storage-file-list-item';
+import {
+  isDirectory,
+  getIconForStorageType,
+  makeFilesCollectionLink,
+} from 'core-app/shared/components/storages/functions/storages.functions';
+
+@Directive()
+export abstract class FilePickerBaseModalComponent extends OpModalComponent implements OnInit, OnDestroy {
+  public breadcrumbs:BreadcrumbsContent;
+
+  public listItems$:Observable<StorageFileListItem[]>;
+
+  public readonly loading$ = new BehaviorSubject<boolean>(true);
+
+  protected get storageLink():IHalResourceLink {
+    return this.locals.storageLink as IHalResourceLink;
+  }
+
+  protected readonly storageFiles$ = new BehaviorSubject<IStorageFile[]>([]);
+
+  private loadingSubscription:Subscription;
+
+  protected constructor(
+    @Inject(OpModalLocalsToken) public locals:OpModalLocalsMap,
+    readonly elementRef:ElementRef,
+    readonly cdRef:ChangeDetectorRef,
+    protected readonly sortFilesPipe:SortFilesPipe,
+    protected readonly storageFilesResourceService:StorageFilesResourceService,
+  ) {
+    super(locals, cdRef, elementRef);
+  }
+
+  ngOnInit():void {
+    super.ngOnInit();
+
+    this.breadcrumbs = new BreadcrumbsContent([{
+      text: this.locals.storageName as string,
+      icon: getIconForStorageType(this.locals.storageType as string),
+      navigate: () => this.changeLevel(null, this.breadcrumbs.crumbs.slice(0, 1)),
+    }]);
+
+    this.listItems$ = this.storageFiles$
```

Comment:
You can just set this as the attribute right away, no need to do it in `  ngOnInit`

Reply:
moved this and the breadcrumbs initialization to the attribute